artifactory_cleanup/rules/delete.py

```diff
@@ -102,3 +103,23 @@ def filter(self, artifacts):
         repositories = utils.build_repositories(artifacts)
         folders = utils.get_empty_folders(repositories)
         return folders
+
+
+class DeleteByRegex(Rule):
+    """
+    Remove artifacts by regex pattern.
```

Comment:
Does it affect only `name` or the whole `path`? 
I think we could explicitly name it `DeleteByRegexpName` to highlight that it works only in `name` attribute